### ISSUE TITLE
[FEAT] Add gradient/activation checkpointing for pretraining methods

### ISSUE THREAD

**Issue by XAheli** [lightly-ai/lightly-train#929]
Pretraining methods like DINOv2 and DINOv3.1 run full forward passes through all transformer blocks in the student backbone, with each block's activations held in memory for the backward pass. For
  ViT-L (24 blocks, embed_dim=1024) and ViT-G (40 blocks, embed_dim=1536), this causes high peak GPU memory usage that limits batch size and model scale on commodity hardware.

Currently there is zero usage of `torch.utils.checkpoint` anywhere in the codebase

The specific hot paths are the block iteration loops that carry gradients during training:

  - **DINOv2 ViT** (`src/lightly_train/_models/dinov2_vit/dinov2_vit_src/models/vision_transformer.py`):
    - `forward_features` (line 350): `for blk in self.blocks: x = blk(x)`
    - `forward_features_list` (line 326): `for blk in self.blocks: x = blk(x)`

  - **DINOv3 ViT** (`src/lightly_train/_models/dinov3/dinov3_src/models/vision_transformer.py`):
    - `forward_features_list` (line 264): `for _, blk in enumerate(self.blocks): x = blk(x, rope_sincos)`
    - `_get_intermediate_layers_not_chunked` (line 317): `for i, blk in enumerate(self.blocks): x = blk(x, rope_sincos)`
    
  - **ECViT** (`src/lightly_train/_models/ecvit/ecvit.py`):
    - `forward_with_grid` (line 415): `for i, blk in enumerate(self.blocks): x = blk(x, rope_sincos=rope_sincos)`

  Teacher forward passes (e.g., `DINOv2._forward_teacher` at `dinov2.py:399`, `DistillationV3._forward_teacher` at `distillationv3.py:293`) run under `@torch.no_grad()` and do not need checkpointing.

  ### Approach
  
  Add an opt-in `activation_checkpointing: bool = False` config option that wraps each transformer block call with `torch.utils.checkpoint.checkpoint(..., use_reentrant=False)` during the student forward pass.
  
cc @liopeer

**Comment by liopeer** [comment]
That looks like the right approach! For the public interface however, I would not go for just a bool value, but leave room for extension later (e.g. custom policies and all that). I would define a config class, maybe starting now with the possibility of only activating it for some blocks:
```python
  class ActivationCheckpointingArgs(PydanticConfig):
      enabled: bool = False
      every_n_blocks: int = 1
```
Users should be able to just pass in `activation_checkpointing_args` as a dictionary.
